### PR TITLE
[Behat] Test for assigning main taxon on new product

### PR DESCRIPTION
**Pull request by stefandoorn**
| Q               | A
| --------------- | -----
| Branch?         | 1.2 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Deprecations?   | no/
| License         | MIT

Was running into this while trying to test for choosing a taxon when creating a new product:

```
  Scenario: Creating a product with a custom slug
    Given I want to create a new simple product
    When I specify its code as "BOARD_MANSION_OF_MADNESS"
    And I choose main taxon "Mugs"
      Behat\Testwork\Call\Exception\FatalThrowableError: Fatal error: Call to undefined method Sylius\Behat\Page\Admin\Product\CreateSimpleProductPage::selectMainTaxon() in vendor/sylius/sylius/src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php:565
```

Not sure if `master` branch is correct btw.

### CODE REVIEW

**Comment by stefandoorn** [comment]
Facing the issue that after the step `When I choose main taxon "Sad"`, each time the element is not visible (e.g. `And I specify its code as "WHISKEY_GENTLEMEN"`). How can I reset that? Also facing this issue while developing a plugin.

**Comment by Zales0123** [comment]
@stefandoorn I've committed some tricky changes to your PR, I hope you don't mind, they should fix the build :)

**Comment by stefandoorn** [comment]
Thanks! 🎉 Should we can also move that AutocompleteHelper to the `UpdateConfigurableProductPage::selectMainTaxon`?

**Comment by pamil** [comment]
Rebased to 1.2 (adding a missing test looks like a bugfix for me) and updated with scalar typehints (as they were introduced by @Zales0123 in the other PR in the meantime). I guess the last comment can be solved by another PR :)

**Comment by stefandoorn** [comment]
Ok :) LGTM.

File: src/Sylius/Behat/Page/Admin/Product/CreateSimpleProductPage.php [hunk 1]
```diff
@@ -223,11 +243,17 @@ protected function getDefinedElements(): array
...
         ]);
     }
 
-    private function selectElementFromAttributesDropdown(string $id): void
+    private function openTaxonBookmarks(): void
+    {
+        $this->getElement('taxonomy')->click();
+    }
+
+    private function selectElementFromAttributesDropdown(int $id): void
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/Product/CreateSimpleProductPage.php]: For some reason, changing `$id` type from `string` to `int` breaks the step :( I would leave it as it was and try to debug it later

**Comment by Zales0123** [comment]
Thank you, Stefan! :1st_place_medal:

**Pull request merged**